### ISSUE TITLE
Does not build, not from AUR or from git clone source with script/build

### ISSUE THREAD

**Issue by Jieiku** [atom-community/atom#478]
```sh
paru atom-community-git
```

```
1 aur/atom-community-git 1.63.0_dev_a2bae55a7-1 [+2 ~0.79]
    A hackable text editor for the 21st century, built on Electron, and based on everything we love about our favorite editors.
:: Packages to install (eg: 1 2 3, 1-3):
:: 1
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

Aur (1) atom-community-git-1.63.0_dev_a2bae55a7-1
```

it eventually errors out with:

```
apm  2.6.5-atomic.1.0
npm  6.14.17
node 12.18.3 x64
atom unknown
python 3.10.8
git 2.38.1
Installing modules ✗
> atom@1.63.0-dev preinstall /home/jieiku/.cache/paru/clone/atom-community-git/src/atom
> node -e 'process.exit(0)'


npm ERR! code E500
npm ERR! 500 Internal Server Error - GET https://www.atom.io/api/packages/language-shellscript/versions/0.28.2/tarball

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jieiku/.atom/.apm/_logs/2022-11-15T19_55_05_486Z-debug.log
Error: Command failed: /home/jieiku/.cache/paru/clone/atom-community-git/src/atom/apm/node_modules/atom-package-manager/bin/apm install
    at checkExecSyncError (node:child_process:861:11)
    at Object.execFileSync (node:child_process:896:15)
    at module.exports (/home/jieiku/.cache/paru/clone/atom-community-git/src/atom/script/lib/run-apm-install.js:12:16)
    at bootstrap (/home/jieiku/.cache/paru/clone/atom-community-git/src/atom/script/bootstrap.js:56:3)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
==> ERROR: A failure occurred in prepare().
    Aborting...
error: failed to build 'atom-community-git-1.63.0_dev_a2bae55a7-1': 
error: packages failed to build: atom-community-git-1.63.0_dev_a2bae55a7-1
```

**Comment by Jieiku** [comment]
I just tried building from source without the AUR, but it is the same error.

**Comment by Jieiku** [comment]
I am guessing this is related: https://github.com/atom-community/atom/issues/459

**Comment by Jieiku** [comment]
The main atom repository has already fixed some of these issues: https://github.com/atom/atom/commit/2291b59a289bc2d9d3bd64482b2964eaf0cb5e11

These changes need to be pulled into atom-community.

For now I am going to clone the main atom github and build from source there.

**Comment by jonian** [comment]
@Jieiku try with this change on the atom main repository https://github.com/lunar-editor/lunar/commit/5bb6d649632890f6598d7e7adb09591b15328bdb

**Comment by DeeDeeG** [comment]
Hi @Jieiku.

#### Issue 1, fetching atom.io URLs is unreliable

For the atom.io URLs, yes, the atom.io server is known to be flaky for several months now. I could have sworn that was fixed here, but I guess not,

The atom.io URLs can be replaced with github.com URLs.

If you do `curl -L --verbose curl -L --verbose https://www.atom.io/api/packages/autocomplete-atom-api/versions/0.10.7/tarball`, for example, you can see it redirects to here: `https://codeload.github.com/atom/autocomplete-atom-api/legacy.tar.gz/refs/tags/v0.10.7`.

You can replace the `atom.io` URL with this `codeload.github.com` URL, or something equivalent like this: `https://github.com/atom/autocomplete-atom-api/tarball/v0.10.7` if you prefer (it ultimately redirects to the same `codeload.github.com` URL).

**In short, change atom.io package URLs to fetch from GitHub.com instead. See above for details.**

#### Issue 2, symbol dumping is error prone

This gives me issues from time to time as well. I don't use the native code debug symbols personally... If you don't need them, you an comment out or otherwise disable the symbol dumping part of script/build. The patch the commenter above posted works. Or you can comment out the `dumpSymbols()` function in `script/build` entirely.

https://github.com/atom-community/atom/blob/a2bae55a76d971507d4b26a34b5aba5f0b8c644c/script/build.js#L218

**In short, disable symbol dumping if you don't need the native code debug symbols for various bundled/core packages.**

**Comment by DeeDeeG** [comment]
FWIW, the AUR is not made by people at this repo, and it is untested by maintainers here. Some effort is made to troubleshoot with people, but it's not 100% top priority, and I think the AUR is not in great shape last I heard (which was several months ago). It could be better now, I guess? (Not having looked at it myself.)

**Comment by Jieiku** [comment]
appreciate the help guys, got atom installed now :)
